Ajout d'un lexique au thème 2

samayala22 opened the pull request.

samayala22 committed:
ajout du lexique

samayala22 committed:
Update theme2.ipynb

- Ajout lexique
- Modif de quelques définitions
- Ajout qq emojis

samayala22 committed:
Merge branch 'gch-faps:main' into theme2-lexique

bibeauv commented on theme2/theme2.ipynb:
```diff
@@ -825,6 +827,45 @@
+    "\n",
+    "- `np.array(<objet>, dtype=)`: création d'un `ndarray` à partir d'une liste python. L'argument `dtype` permet de spécifier le [type](https://numpy.org/doc/stable/user/basics.types.html) de données numériques qui sera employé. \n",
+    "- `np.arange(<debut>, <fin>, <pas>)`: génère les points à partir d'un pas $\\Delta T$ spécifique. Attention, la valeur de fin est exclue. \n",
+    "- `np.linspace(<debut>, <fin>, <nb de points>)`: génère le nombre de points voulus espacés linéairement\n",
```
```suggestion
    "- `np.linspace(<debut>, <fin>, <nb de points>)`: génère le nombre de points voulus espacés linéairement. Ici, la valeur de fin est inclue.\n",
```

bibeauv commented on theme2/theme2.ipynb:
```diff
@@ -825,6 +827,45 @@
+    "\n",
+    "### ✔️ Vu dans l'exemple 3\n",
+    "\n",
+    "- `np.zeros(<dimensions>, dtype=)`: création d'un `ndarray` de zeros, utile pour l'initialisation de vecteurs, matrices que l'on va remplir plus tard.\n",
```
```suggestion
    "- `np.zeros(<dimensions>, dtype=)`: création d'un `ndarray` de zeros, utile pour l'initialisation de vecteurs ou de matrices qui seront remplis subséquemment dans un code.\n",
```

bibeauv commented on theme2/theme2.ipynb:
```diff
@@ -825,6 +827,45 @@
+    "- `np.array(<objet>, dtype=)`: création d'un `ndarray` à partir d'une liste python. L'argument `dtype` permet de spécifier le [type](https://numpy.org/doc/stable/user/basics.types.html) de données numériques qui sera employé. \n",
+    "- `np.arange(<debut>, <fin>, <pas>)`: génère les points à partir d'un pas $\\Delta T$ spécifique. Attention, la valeur de fin est exclue. \n",
+    "- `np.linspace(<debut>, <fin>, <nb de points>)`: génère le nombre de points voulus espacés linéairement\n",
+    "- `ndarray.size`: retourne le nombre total d'éléments dans un array\n",
```
```suggestion
    "- `ndarray.size`: retourne le nombre total d'éléments dans un `ndarray`.\n",
```

bibeauv commented on theme2/theme2.ipynb:
```diff
@@ -825,6 +827,45 @@
+    "\n",
+    "### ✔️ Vu dans l'exemple 4\n",
+    "\n",
+    "- `ndarray.reshape(<dimensions>)`: changer la forme du array.\n",
```
```suggestion
    "- `ndarray.reshape(<dimensions>)`: changer la forme d'un `ndarray` selon les nouvelles dimensions spécifiées.\n",
```

bibeauv commented on theme2/theme2.ipynb:
```diff
@@ -825,6 +827,45 @@
+    "### ✔️ Vu dans l'exemple 4\n",
+    "\n",
+    "- `ndarray.reshape(<dimensions>)`: changer la forme du array.\n",
+    "- `ndarray.flatten()`: ramène le array à un vecteur 1D.\n",
```
```suggestion
    "- `ndarray.flatten()`: ramène le `ndarray` à un vecteur 1D.\n",
```

bibeauv commented on theme2/theme2.ipynb:
```diff
@@ -825,6 +827,45 @@
+    "\n",
+    "- `ndarray.reshape(<dimensions>)`: changer la forme du array.\n",
+    "- `ndarray.flatten()`: ramène le array à un vecteur 1D.\n",
+    "- `np.eye(<nb lignes>, <nb colonnes>, dtype=)`: création d'une matrice avec des 1 sur la diagonale. Utiliser `np.eye(<n>)` pour une matrice carrée nxn. \n"
```
```suggestion
    "- `np.eye(<nb lignes>, <nb colonnes>, dtype=)`: création d'une matrice identité, c'est-à-dire avec des 1 sur la diagonale. Utiliser `np.eye(<n>)` pour une matrice carrée n x n. \n"
```

bibeauv commented on theme2/theme2.ipynb:
```diff
@@ -825,6 +827,45 @@
+    "- `np.linalg.inv(A)`: inverse d'une matrice A\n",
+    "- Opérateur `@` pour la multiplication matricielle\n",
+    "\n",
+    "### ✔️ Vu dans l'exemple 3\n",
```
Pour tout ce qui est `<dimensions>`, il faut faire attention à utiliser un tuple (2,3) pour une matrice. Si on veut créer une matrice de dimension (2,3), il faut faire : `np.zeros((2,3))` et non `np.zeros(2,3)`, sinon ça throw une erreur.
Donc peut-être dire qu'habituellement, les dimensions sont toujours en tuple. Et pour `np.reshape`, je pense que les deux se font. Tu peux utiliser un tuple et tu peux spécifier aussi le nombre de lignes et le nombre de colonnes séparément dans 2 flags différents. Avec `np.eye`, tu expliques justement très bien la distinction !

samayala22 replied:
J'ai ajouté un commentaire pour clarifier l'ambiguïté. 👍

bibeauv commented on theme2/theme2.ipynb:
```diff
@@ -825,6 +827,45 @@
    "source": [
     "- Il est recommandé d'avoir le python shell ouvert sur un terminal pour pouvoir facilement tester ces fonctions de manipulations matricielles lors de l'écriture de code."
    ]
+  },
```
C'est vraiment juste une question d'esthétique, mais desfois tu termines tes _bullet points_ avec un point, desfois non 😄

samayala22 replied:
Ouaip je vais régler ça 😅

bibeauv reviewed (changes requested):
Vraiment bien le lexique, je trouve que ça fait vraiment un bon résumé ! :)

samayala22 committed:
Update theme2/theme2.ipynb

Co-authored-by: Valérie Bibeau <50414572+bibeauv@users.noreply.github.com>

samayala22 committed:
Update theme2/theme2.ipynb

Co-authored-by: Valérie Bibeau <50414572+bibeauv@users.noreply.github.com>

samayala22 committed:
Update theme2/theme2.ipynb

Co-authored-by: Valérie Bibeau <50414572+bibeauv@users.noreply.github.com>

samayala22 committed:
Update theme2/theme2.ipynb

Co-authored-by: Valérie Bibeau <50414572+bibeauv@users.noreply.github.com>

samayala22 committed:
Update theme2/theme2.ipynb

Co-authored-by: Valérie Bibeau <50414572+bibeauv@users.noreply.github.com>

samayala22 committed:
Update theme2/theme2.ipynb

Co-authored-by: Valérie Bibeau <50414572+bibeauv@users.noreply.github.com>

samayala22 committed:
Update theme2.ipynb

- Clarification des notions
- Ajout des sections librairies et installation
- Modification des éléments suggérés par Valérie.

samayala22 committed:
Merge branch 'main' into theme2-lexique

samayala22 committed:
pre-commit hook first run

bibeauv reviewed (approved):
Excellent !
